pennylane/drawer/tape_text.py

```diff
@@ -89,6 +160,60 @@ def _add_measurement(m, layer_str, wire_map, decimals, cache):
     return layer_str
 
 
+def _find_mid_measure_cond_connections(operations, layers):
+    """Create dictionaries with various mappings needed for drawing
+    mid-circuit measurements and classically controlled operators
+    correctly."""
+
+    # Map between mid-circuit measurements and their position on the drawing
+    # The bit map only contains mid-circuit measurements that are used in
+    # classical conditions.
+    bit_map = {}
+
+    # Map between classical bit positions and whether we have reached the mid-circuit
+    # measurement of that bit. This is needed to know when to start drawing the bit line.
+    bit_measurements_reached = []
+
+    # Map between classical bit positions and the final layer where the bit is used.
+    # This is needed to know when to stop drawing a bit line. The bit is the index,
+    # so each of the two lists must have the same length as the number of bits
+    all_bit_terminal_layers = [[], []]
+
+    measurements_for_conds = set()
+    conditional_ops = []
+    for op in operations:
+        if op.__class__.__name__ == "Conditional":
+            measurements_for_conds.update(op.meas_val.measurements)
+            conditional_ops.append(op)
+
+    if len(measurements_for_conds) > 0:
+        cond_mid_measures = [
+            op for op in operations if isinstance(op, MidMeasureMP) if op in measurements_for_conds
+        ]
+        cond_mid_measures.sort(
+            key=lambda mp: operations.index(mp)  # pylint: disable=unnecessary-lambda
+        )
+
+        bit_map = dict(zip(cond_mid_measures, range(len(cond_mid_measures))))
+
+        n_bits = len(bit_map)
+        bit_measurements_reached = [False for _ in range(n_bits)]
```

Comment:
Is this ever not going to be `False`? I'm a little confused by what this variable indicates.

Reply:
It get's changed to True at the end of the loop if the current layer contained a mid-circuit measurement. This is to keep track of when to start drawing the bit line. The `bit_terminal_layers` list keeps track of the last layer where bit lines must be drawn.

Reply:
Removed this.